File: solutions/typescript/resistor-color-trio/1/resistor-color-trio.ts
export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
];

export function decodedResistorValue(colors: string[]): string {
  // Get the first two colors for the main value
  const firstValue = COLORS.indexOf(colors[0]);
  const secondValue = COLORS.indexOf(colors[1]);
  const mainValue = firstValue * 10 + secondValue;
  
  // Get the multiplier (number of zeros) from the third color
  const zeros = COLORS.indexOf(colors[2]);
  
  // Calculate the total resistance
  let resistance = mainValue * Math.pow(10, zeros);
  
  // Determine the appropriate metric prefix
  const prefixes: { value: number; name: string }[] = [
    { value: 1000000000, name: 'giga' },
    { value: 1000000, name: 'mega' },
    { value: 1000, name: 'kilo' }
  ];
  
  for (const prefix of prefixes) {
    if (resistance >= prefix.value && resistance % prefix.value === 0) {
      resistance = resistance / prefix.value;
      return `${resistance} ${prefix.name}ohms`;
    }
  }
  
  return `${resistance} ohms`;
}

File: solutions/ruby/resistor-color-trio/1/resistor_color_trio.rb
class ResistorColorTrio
  COLORS = %w[black brown red orange yellow green blue violet grey white].freeze
  
  # Definimos las unidades en potencias de 1000
  UNITS = %w[ohms kiloohms megaohms gigaohms].freeze

  def initialize(colors)
    @colors = colors
  end

  def label
    # 1. Obtener valores numéricos de las bandas
    v1 = COLORS.index(@colors[0])
    v2 = COLORS.index(@colors[1])
    exponent = COLORS.index(@colors[2])

    # 2. Calcular valor base en ohms
    value = (v1 * 10 + v2) * (10**exponent)

    # 3. Escalar el valor y la unidad
    unit_index = 0
    # Mientras el valor sea mayor o igual a 1000 y tengamos más unidades
    while value >= 1000 && unit_index < UNITS.length - 1
      value /= 1000
      unit_index += 1
    end

    "Resistor value: #{value} #{UNITS[unit_index]}"
  end
end